src/Hack.sol

```diff
@@ -602,13 +602,16 @@ contract GuessTheAverage {
     /// @param _count The number of transactions to execute. Executes until the end if set to "0" or number higher than number of winners in the list.
     function distribute(uint256 _count) public {
         require(currentStage == Stage.WinnersFound, "Winners must have been found");
-        for (uint256 i = cursorDistribute; i < winners.length && (_count == 0 || i < cursorDistribute + _count); i++) {
-            // Send ether to the winners, use send not to block, if one of the account cannot receive ETH.
-            winners[i].call{value: totalBalance / (winners.length - numberOfLosers)}("");
-            if (i == winners.length - 1) currentStage = Stage.Distributed;
+
+        while (cursorDistribute < winners.length && _count != 0) {
+            // Send ether to the winners. Do not block if one of the account cannot receive ETH.
+            winners[cursorDistribute++].call{value: totalBalance / (winners.length - numberOfLosers)}("");
```

Comment:
that's not ideal in term of storage writing, could be better if `cursorDistribute` is stored in a temp variable?

Reply:
ok haha let's go